src/json.rs

```diff
@@ -49,7 +59,14 @@ impl JsonValue {
     pub fn u64(&self, key: &str) -> Result<u64, JsonError> {
         match self {
             JsonValue::Object(o) => match o.get(key).ok_or(JsonError::MissingValue)? {
-                JsonValue::Number(n) => Ok(*n as u64),
+                JsonValue::Number(n) => {
+                    // Upper bound is covered by the type limitation
+                    if (*n as i128) < u64::MIN as i128 {
```

Comment:
Why is this conversion `i128`? Shouldn't this just be `u64` ...?

Reply:
It's overflow check

Reply:
I get that, but why would you check it `as i128`